src/lib/y2network/proposal_settings.rb

```diff
@@ -0,0 +1,121 @@
+  class ProposalSettings
+    include Yast::Logger
+    include Yast::I18n
+
+    # @return [Boolean] network service to be used after the installation
+    attr_accessor :backend
+
+    # Constructor
+    def initialize
+      Yast.import "ProductFeatures"
+      Yast.import "PackagesProposal"
+      Yast.import "Lan"
+
+      @backend = use_network_manager? ? :network_manager : :wicked
+    end
+
+    # Adds the NetworkManager package to the {Yast::PackagesProposal} and sets
+    # NetworkManager as the backend to be used
+    def enable_network_manager!
+      Yast::PackagesProposal.AddResolvables("network", :package, ["NetworkManager"])
+      Yast::PackagesProposal.RemoveResolvables("network", :package, ["wicked"])
+
+      log.info "Enabling NetworkManager"
+      self.backend = :network_manager
+    end
+
+    # Add the wicked package to the {Yast::PackagesProposal} and sets wicked
+    # as the backend to be used
+    def enable_wicked!
+      Yast::PackagesProposal.AddResolvables("network", :package, ["wicked"])
+      Yast::PackagesProposal.RemoveResolvables("network", :package, ["NetworkManager"])
+
+      log.info "Enabling Wicked"
+      self.backend = :wicked
+    end
+
+    # Convenience method to obtain whether the NetworkManager package is
+    # available or not.
+    #
+    # @return [Boolean] false if no package available, true otherwise
+    def network_manager_available?
+      p = Y2Packager::Package.find("NetworkManager").first
+      return false if p.nil?
+      log.info("The NetworkManager package status: #{p.status}")
+      true
+    end
+
+    class << self
+      # Singleton instance
+      def instance
+        @instance ||= create_instance
+      end
+
+      # Enforce a new clean instance
+      def create_instance
+        @instance = new
+      end
+
+      # Make sure only .instance and .create_instance can be used to
+      # create objects
+      private :new, :allocate
+    end
+
+  private
+
+    # Convenienve method that verify if Network Manager should be used or not
+    # according to the control file defaults and package availability.
+    #
+    # @return [Boolean] true if should be used; false otherwise
+    def use_network_manager?
+      return false unless network_manager_available?
+
+      network_manager_default?
+    end
+
+    # Determine whether NetworkManager should be selected by default according
+    # to the product control file
+    #
+    # @return [Boolean] true if NM should be enabled; false otherwise
+    def network_manager_default?
+      case Yast::ProductFeatures.GetStringFeature("network", "network_manager")
+      when ""
+        # compatibility: use the boolean feature
+        # (defaults to false)
+        Yast::ProductFeatures.GetBooleanFeature("network", "network_manager_is_default")
+      when "always"
+        true
+      when "laptop"
+        laptop = Arch.is_laptop
+        log.info("Is a laptop: #{laptop}")
```

Comment:
I'd use a more meaningful message here, something similar to

`NetworkManager is selected by default because the installation is running in a laptop`

However, I am wondering why it worth to log specific information for this case and not for the rest,

`NetworkManager is not selected by default`
`Product control file determines to select always NetworkManager as default network backend`,

What am I missing?

Reply:
That only logs if it is a laptop or not, when the control file "network_manager" option is "laptop", so basically when it is a laptop it will be enabled by default but in case of a server or workstation it will not.

Reply:
Thank you for the explanation, @teclator 

To be honest, I did understand that, but I do not know why this case is more special that the other ones. Additionally, I think it will weird to find `Is a laptop: true` in the logs, without more contextual information (apart from the file and line number :P)

Reply:
Well, the logs are closed together, so when you are debugging it will be easy to identity as we are currently logging if NM will be used by default or not.